packages/protocol/test/liquidity/grandamento.ts

```diff
@@ -0,0 +1,1207 @@
+        await grandaMento.cancelExchangeProposal(1, { from: alice })
+        assertEqualBNArray(await grandaMento.getActiveProposalIds(), [new BigNumber(3)])
+        // Remove 3
+        await grandaMento.cancelExchangeProposal(3, { from: alice })
+        assertEqualBNArray(await grandaMento.getActiveProposalIds(), [])
+      })
+
+      describe('when selling the stable token', () => {
+        beforeEach(async () => {
+          await grandaMento.createExchangeProposal(
+            stableTokenRegistryId,
+            stableTokenSellAmount,
+            false,
+            {
+              from: alice,
+            }
+          )
+        })
+
+        it('refunds the same stable token amount as the original deposit when the inflation factor is 1', async () => {
+          const grandaMentoBalanceBefore = await stableToken.balanceOf(grandaMento.address)
+          const aliceBalanceBefore = await stableToken.balanceOf(alice)
+          await grandaMento.cancelExchangeProposal(1, { from: alice })
+          const grandaMentoBalanceAfter = await stableToken.balanceOf(grandaMento.address)
+          const aliceBalanceAfter = await stableToken.balanceOf(alice)
+
+          assertEqualBN(
+            grandaMentoBalanceBefore.minus(grandaMentoBalanceAfter),
+            stableTokenSellAmount
+          )
+          assertEqualBN(aliceBalanceAfter.minus(aliceBalanceBefore), stableTokenSellAmount)
+        })
+
+        it('refunds the appropriate stable token amount value when the inflation factor is not 1', async () => {
+          const inflationFactor = 1.1
+          await stableToken.setInflationFactor(toFixed(inflationFactor))
+
+          const grandaMentoBalanceBefore = await stableToken.balanceOf(grandaMento.address)
+          const aliceBalanceBefore = await stableToken.balanceOf(alice)
+          await grandaMento.cancelExchangeProposal(1, { from: alice })
+          const grandaMentoBalanceAfter = await stableToken.balanceOf(grandaMento.address)
+          const aliceBalanceAfter = await stableToken.balanceOf(alice)
+
+          const valueAmount = unitsToValue(stableTokenSellAmount, inflationFactor)
+          assertEqualBN(grandaMentoBalanceBefore.minus(grandaMentoBalanceAfter), valueAmount)
+          assertEqualBN(aliceBalanceAfter.minus(aliceBalanceBefore), valueAmount)
+        })
+
+        it('refunds the entire balance if the amount to refund is higher than the balance', async () => {
+          const newGrandaMentoBalance = unit.times(400)
+          // Remove some stableToken from granda mento to artificially simulate a situation
+          // where the refund amount is > granda mento's balance.
+          await stableToken.transferFrom(
+            grandaMento.address,
+            owner,
+            stableTokenSellAmount.minus(newGrandaMentoBalance)
+          )
+
+          const aliceBalanceBefore = await stableToken.balanceOf(alice)
+          await grandaMento.cancelExchangeProposal(1, { from: alice })
```

Comment:
would love to see a lot of this and tests below deduped
1200 lines of tests seems like a lot for a fairly simple contract

Reply:
Deduped a lot of tests-- still a pretty big file, but marginally better now